[TestWebKitAPI] Fix the public iOS 26 build

emw-apple opened the pull request:
#### 77271dc9c550b48c85902fe620b67fa39d6191d1
<pre>
[TestWebKitAPI] Fix the public iOS 26 build
<a href="https://bugs.webkit.org/show_bug.cgi?id=300036">https://bugs.webkit.org/show_bug.cgi?id=300036</a>

Reviewed by Abrar Rahman Protyasha.

Update UITextInput protocol in SPI headers (this protocol is actually
API, but WKWebView still uses a pre-API internal protocol). This needs
to be visible from Swift, so add it to the bridging header. Unlike our
frameworks, the test bundle is not itself a module, so we do not need to
modularize UIKitSPIForTesting.h and friends to be able to use it from
swift code.

Along the way, fix a couple of public build issues, noted below.

Test: Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h

Test: Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h

* Configurations/CommonBase.xcconfig: Change the definition of
  SDKDB_HAS_148943382 to be internal only. When we generate partial
  SDKDBs from the public SDK, they do not exhibit the same bug. This
  change reverts an earlier attempt to work around the bug in
  300877@main.

* Source/WebCore/Configurations/AllowedSPI-legacy.toml: Revert an
  earlier attempt to fix the above problem (300877@main).

* Tools/TestRunnerShared/spi/UIKitSPIForTesting.h:
* Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h:

* WebKitLibraries/SDKs/iphoneos26.0-additions.sdk/System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore.tbd:
  Removed. The public build gets UIKit symbols from UIKit.tbd directly,
  not by binding to UIKitCore. This sidesteps an issue where WebKitSwift
  binds to different UIKit symbols in debug and release configurations,
  which confuses extract-tbds-from-internal-sdk.
* WebKitLibraries/SDKs/iphonesimulator26.0-additions.sdk/System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore.tbd: Removed.

Canonical link: <a href="https://commits.webkit.org/300969@main">https://commits.webkit.org/300969@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/29c226ee8cc4c410bb51ce7275313ab0ecd89ebb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124503 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44181 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34915 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131341 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/76483 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d3d0ed98-fcaa-406c-b685-91d29a81f949) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126380 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44898 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52762 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/94712 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/76483 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c41bc27b-99ed-4bd3-b393-e3b6644ec80f) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127457 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/35789 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/111359 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75289 "Passed tests") | | [❌ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/bccf9e1a-7957-4784-9d02-9d22d73c5501) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/34727 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/29515 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74822 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/105541 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/29747 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134006 "Built successfully") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/51374 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/39199 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/103187 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51781 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/107583 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102972 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26210 "Built successfully and passed tests") | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/48337 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/26609 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/48317 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/51244 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57034 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/50655 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/54014 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/52335 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/59bf748982ff268fe5cc9a1085784a2da32fb654. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51690

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/50d126395dd551433fe4f3935506dbd0fef72ca7. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51690

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/29c226ee8cc4c410bb51ce7275313ab0ecd89ebb. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51690

emw-apple committed:
[TestWebKitAPI] Fix the public iOS 26 build
https://bugs.webkit.org/show_bug.cgi?id=300036

Reviewed by Abrar Rahman Protyasha.

Update UITextInput protocol in SPI headers (this protocol is actually
API, but WKWebView still uses a pre-API internal protocol). This needs
to be visible from Swift, so add it to the bridging header. Unlike our
frameworks, the test bundle is not itself a module, so we do not need to
modularize UIKitSPIForTesting.h and friends to be able to use it from
swift code.

Along the way, fix a couple of public build issues, noted below.

Test: Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h

Test: Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h

* Configurations/CommonBase.xcconfig: Change the definition of
  SDKDB_HAS_148943382 to be internal only. When we generate partial
  SDKDBs from the public SDK, they do not exhibit the same bug. This
  change reverts an earlier attempt to work around the bug in
  300877@main.

* Source/WebCore/Configurations/AllowedSPI-legacy.toml: Revert an
  earlier attempt to fix the above problem (300877@main).

* Tools/TestRunnerShared/spi/UIKitSPIForTesting.h:
* Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h:

* WebKitLibraries/SDKs/iphoneos26.0-additions.sdk/System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore.tbd:
  Removed. The public build gets UIKit symbols from UIKit.tbd directly,
  not by binding to UIKitCore. This sidesteps an issue where WebKitSwift
  binds to different UIKit symbols in debug and release configurations,
  which confuses extract-tbds-from-internal-sdk.
* WebKitLibraries/SDKs/iphonesimulator26.0-additions.sdk/System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore.tbd: Removed.

Canonical link: https://commits.webkit.org/300969@main
